tests/test-search-engine.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Class SearchEngine
+ *
+ * @package Unlist_Posts
+ */
+
+/**
+ * Make sure unlisted posts are hidden from search engine.
+ */
+class SearchEngine extends WP_UnitTestCase {
+
+	/**
+	 * User ID for a editor user..
+	 *
+	 * @var int
+	 */
+	private $editor_user_id;
+
+	/**
+	 * Setup the tests class.
+	 *
+	 * @return void
+	 */
+	public function setUp() {
+		$this->editor_user_id = self::factory()->user->create(
+			array(
+				'role' => 'editor',
+			)
+		);
+
+		_delete_all_posts();
+	}
+
+	/**
+	 * Check search engine robots tag for unlisted post.
+	 */
+	public function test_unlisted_noindex() {
+		wp_set_current_user( $this->editor_user_id );
+
+		// Create an unlisted post.
+		$unlisted_post = self::factory()->post->create();
+
+		// Set the post as unlisted.
+		$_POST['unlist_posts']       = true;
+		$_POST['unlist_post_nounce'] = wp_create_nonce( 'unlist_post_nounce' );
+		Unlist_Posts_Admin::instance()->save_meta( $unlisted_post );
+
+		// Set unlisted post as a current post.
+		$GLOBALS['post'] = get_post( $unlisted_post );
+
+		// Get wp_robots string of a current post.
+		ob_start();
+		wp_robots();
+		$output = ob_get_clean();
+
+		// Unlisted post should contain noindex string in meta.
+		$this->assertContains( 'noindex', $output );
+	}
+
+	/**
+	 * Check search engine robots tag for listed post.
+	 */
+	public function test_listed_noindex() {
+		wp_set_current_user( $this->editor_user_id );
+
+		// Create an listed post.
+		$listed_post = self::factory()->post->create();
+
+		// Set listed post as a current post.
+		$GLOBALS['post'] = get_post( $listed_post );
+
+		// Get wp_robots string of a current post.
+		ob_start();
+		wp_robots();
+		$output = ob_get_clean();
+
+		// Listed post should not contain noindex string in meta.
+		$this->assertNotContains( 'noindex', $output );
```

Comment:
```suggestion
		// Listed post should not contain noindex string in meta.
		$this->assertNotContains( 'noindex', get_echo( 'wp_robots' ) );
```

One final suggestion - there is also [`get_echo()`](https://github.com/WordPress/wordpress-develop/blob/b4f354d524ca6e7f9645eb3d5ef3089e0492a19f/tests/phpunit/includes/utils.php#L351-L355) function which does what `ob_start()` logic is doing above.

Reply:
Do check out many other utilities in this file as those can be handy when writing tests